### ISSUE TITLE
 undefined method `make_tmpname' with 

### ISSUE THREAD

**Issue by jas01** [puppetlabs/pdk#455]
I running pdk with arch linux, so I install pdk with gem install pdk.
Some command work fine, but some dont with this message (here it's convert)
`````
[jas01@io java][ldap-auth]$ pdk convert                                                                                                                       
Traceback (most recent call last):                                                                                                                            
        13: from /home/jas01/.gem/ruby/2.5.0/bin/pdk:23:in `<main>'                                                                                           
        12: from /home/jas01/.gem/ruby/2.5.0/bin/pdk:23:in `load'                                                                                             
        11: from /home/jas01/.gem/ruby/2.5.0/gems/pdk-1.4.1/exe/pdk:6:in `<top (required)>'                                                                   
        10: from /home/jas01/.gem/ruby/2.5.0/gems/pdk-1.4.1/lib/pdk/cli.rb:17:in `run'                                                                        
         9: from /home/jas01/.gem/ruby/2.5.0/gems/cri-2.9.1/lib/cri/command.rb:282:in `run'                                                                   
         8: from /home/jas01/.gem/ruby/2.5.0/gems/cri-2.9.1/lib/cri/command.rb:264:in `run'                                                                   
         7: from /home/jas01/.gem/ruby/2.5.0/gems/cri-2.9.1/lib/cri/command.rb:319:in `run_this'                                                              
         6: from /home/jas01/.gem/ruby/2.5.0/gems/pdk-1.4.1/lib/pdk/cli/convert.rb:38:in `block (2 levels) in <module:CLI>'                                   
         5: from /home/jas01/.gem/ruby/2.5.0/gems/pdk-1.4.1/lib/pdk/module/convert.rb:10:in `invoke'                                                          
         4: from /home/jas01/.gem/ruby/2.5.0/gems/pdk-1.4.1/lib/pdk/module/convert.rb:20:in `run'                                                             
         3: from /home/jas01/.gem/ruby/2.5.0/gems/pdk-1.4.1/lib/pdk/module/convert.rb:69:in `stage_changes!'                                                  
         2: from /home/jas01/.gem/ruby/2.5.0/gems/pdk-1.4.1/lib/pdk/module/convert.rb:69:in `new'                                                             
         1: from /home/jas01/.gem/ruby/2.5.0/gems/pdk-1.4.1/lib/pdk/module/templatedir.rb:50:in `initialize'                                                  
/home/jas01/.gem/ruby/2.5.0/gems/pdk-1.4.1/lib/pdk/util.rb:35:in `make_tmpdir_name': undefined method `make_tmpname' for Dir::Tmpname:Module                  
(NoMethodError)  

```

**Comment by scotje** [comment]
Looks like Ruby 2.5 removed that method from `Dir::Tmpname`:

https://github.com/ruby/ruby/commit/25d56ea7b7b52dc81af30c92a9a0e2d2dab6ff27

**Comment by scotje** [comment]
@jas01 for the moment if you have the option of running under Ruby 2.4 that will be easiest. The fix for this issue is pretty simple but there are a few other bits missing for Ruby 2.5 that will take a little longer.

**Comment by jas01** [comment]
Le 21/03/2018 à 19:01:46+0000, Jesse Scott a écrit
> @jas01 for the moment if you have the option of running under Ruby 2.4 that
> will be easiest. The fix for this issue is pretty simple but there are a few
> other bits missing for Ruby 2.5 that will take a little longer.

No problem.

It's the first time I use pdk. So I just report a (seem to me) a little bug.

I didn't create so much module for puppet :-( ...so I can wait ;-)

Thanks for the work

Regards

**Comment by scotje** [comment]
This is now captured as https://tickets.puppetlabs.com/browse/PDK-877 in JIRA.

**Comment by scotje** [comment]
Fixed in #459
